### PR TITLE
Make `Tooltip` state class public

### LINKED ISSUE #96127: _TooltipState should probably be public
The doc of TooltipTriggerMode.manual https://api.flutter.dev/flutter/material/TooltipTriggerMode.html said it can only be shown by calling `ensureTooltipVisible` on the _TooltipState. However the class itself is private. Right now users have to cast the state into dynamic to call the `ensureTooltipVisible`

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/96127

The PR follows the same pattern as [RefreshIndicatorState](https://github.com/flutter/flutter/blob/2015143fbf80a1fefd3a222f1d6a264b4d68003d/packages/flutter/lib/src/material/refresh_indicator.dart#L228)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tooltip.dart [hunk 1]
```diff
@@ -295,7 +295,9 @@ class Tooltip extends StatefulWidget {
   }
 }
 
-class _TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
+/// Contains the state for a [Tooltip]. This class can be used to
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:298]: Our style guide suggests:
`The first paragraph of any dartdoc section must be a short self-contained sentence`

**Review by chunhtai — APPROVED** [review]
LGTM, except for one document nit

File: packages/flutter/lib/src/material/tooltip.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/tooltip.dart]: Thanks for the review! Made the changes.

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/100553

**Pull request merged**